routing/pathfind.go

```diff
@@ -644,6 +644,34 @@ func findPath(g *graphParams, r *RestrictParams, cfg *PathFindingConfig,
 				edge.ChannelID)
 		}
 
+		switch {
+
+		// If this edge takes us to the final hop, we'll set the node
+		// features to those determined above. These are either taken
+		// from the destination features, e.g. virual or invoice
+		// features, or loaded as a fallback from the graph. The
+		// transitive dependencies were already validated above, so no
+		// need to do so now.
+		//
+		// NOTE: This may overwrite features loaded from the graph if
+		// destination features were provided. This is fine though,
+		// since our route construction does not care where the features
+		// are actually taken from. In the future we may wish to do
+		// route construction within findPath, and avoid using
+		// ChannelEdgePolicy altogether.
+		case edge.Node.PubKeyBytes == target:
+			edge.Node.Features = features
+
+		// Otherwise, this is some other intermediary node. Verify the
+		// transitive feature dependencies for this node, and skip the
+		// channel if they are invalid.
+		default:
+			err := feature.ValidateDeps(edge.Node.Features)
```

Comment:
This is potentially executed for every incoming edge of a node, while it really only need to be done once. Probably not an issue for now. Or would you want to keep a separate map for this, mapping `Node` to `Features` with a 'seen' marker?

Reply:
I also need to have those features available at the right time for the running payload size calculation. What do you think of this one? https://github.com/lightningnetwork/lnd/pull/3851 (last commit)

Reply:
Discussed offline, will be dealt with in https://github.com/lightningnetwork/lnd/pull/3841